BLUESPAWN-win-client/src/util/permissions/permissions.cpp

```diff
@@ -224,88 +292,264 @@ namespace Permissions{
 		wName{ name }, wDomainName{ domain }, sdSID{ sid }, bExists{ exists }, otType{ type } {}
 
-	bool Owner::Exists() const{
+	bool Owner::Exists() const {
 		return bExists;
 	}
 
-	std::wstring Owner::GetName() const{
+	std::wstring Owner::GetName() const {
 		return wName;
 	}
 
-	std::wstring Owner::GetDomainName() const{
+	std::wstring Owner::GetDomainName() const {
 		return wDomainName;
 	}
 
-	PSID Owner::GetSID() const{
-		if(otType == OwnerType::USER) return sdSID.GetUserSID();
+	PSID Owner::GetSID() const {
+		if (otType == OwnerType::USER) return sdSID.GetUserSID();
 		return sdSID.GetGroupSID();
 	}
 
-	OwnerType Owner::GetOwnerType() const{
+	OwnerType Owner::GetOwnerType() const {
 		return otType;
 	}
 
-	std::wstring Owner::ToString() const{
+	std::wstring Owner::ToString() const {
 		return wName;
 	}
 
-	User::User(IN const std::wstring& uName) : Owner{ uName , true, OwnerType::USER }{
+	std::vector<std::wstring> Owner:: GetPrivileges() {
+		//Ensure policy handle is initialized
+		if (!bPolicyInitialized) {
+			InitializePolicy();
+			if (!bPolicyInitialized) {
+				LOG_ERROR("Error getting owner privliges, couldn't initialize policy handle.");
+				return std::vector<std::wstring>{ };
+			}
+		}
+		PLSA_UNICODE_STRING pReceivedPrivs{ nullptr };
+		ULONG uPrivCount{ 0 };
+		std::vector<std::wstring> vPrivs{ };
+		auto hr = LsaNtStatusToWinError(LsaEnumerateAccountRights(lPolicyHandle, GetSID(), &pReceivedPrivs, &uPrivCount));
+		AllocationWrapper awReceivedPrivsHandler{ pReceivedPrivs, 0, AllocationWrapper::NET_ALLOC };
+		if (hr != ERROR_SUCCESS && otType != OwnerType::USER) {
+			LOG_ERROR("Error getting owner privileges. (Error: " << GetLastError() << ")");
+			SetLastError(hr);
+			return std::vector<std::wstring>{ };
+		}
+		else if (hr == ERROR_SUCCESS) {
+			for (int i = 0; i < uPrivCount; i++) {
+				vPrivs.emplace_back(LsaUnicodeStringToWString(pReceivedPrivs[i]));
+			}
+		}
+		//Get privileges from groups that a user belongs to
+		if (otType == OwnerType::USER) {
+			PGROUP_USERS_INFO_0 pGroupInfo{ nullptr };
+			DWORD dEntriesRead{ 0 };
+			DWORD dEntriesTotal{ 0 };
+			NET_API_STATUS stat = NetUserGetLocalGroups(wDomainName.c_str(), wName.c_str(), 0, LG_INCLUDE_INDIRECT , reinterpret_cast<LPBYTE *>(&pGroupInfo), MAX_PREFERRED_LENGTH, &dEntriesRead, &dEntriesTotal);
+			AllocationWrapper awGroupInfoHandler{ pGroupInfo, sizeof(GROUP_USERS_INFO_0) * dEntriesRead, AllocationWrapper::NET_ALLOC };
+			if (stat != NERR_Success) {
+				LOG_ERROR("Error getting user groups. (Net Error: " << stat << ")");
+				return vPrivs;
+			}
+			//Add all privileges from groups to list of user's privileges
+			for (int i = 0; i < dEntriesRead; i++) {
+				Owner oGroup{ pGroupInfo[i].grui0_name };
+				hr = LsaNtStatusToWinError(LsaEnumerateAccountRights(lPolicyHandle, oGroup.GetSID(), &pReceivedPrivs, &uPrivCount));
+				awReceivedPrivsHandler = { pReceivedPrivs, 0, AllocationWrapper::NET_ALLOC };
```

Comment:
Codacy found an issue: [Variable 'awReceivedPrivsHandler' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5628317)